packages/components/modules/messages/ChatRoom/ChatRoomHeader/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { FC } from 'react'
+
+import {
+  AvatarWithPlaceholder,
+  IconButton,
+  Iconify,
+  useResponsive,
+} from '@baseapp-frontend/design-system'
+
+import { Box, Typography } from '@mui/material'
+import { useFragment } from 'react-relay'
+
+import { ChatRoomHeaderFragment$data } from '../../../../__generated__/ChatRoomHeaderFragment.graphql'
+import { useChatRoom } from '../../context'
+import { ChatRoomHeaderFragment } from '../../graphql/queries/ChatRoomHeaderFragment'
+import useNameAndAvatar from '../../hooks/useNameAndAvatar'
+import { getParticipantCount, isGroupChat } from '../../hooks/utils'
+import { ChatHeaderContainer } from './styled'
+import { ChatRoomHeaderProps } from './types'
+
+const getSubtitle = (roomHeader: ChatRoomHeaderFragment$data) => {
+  if (isGroupChat(roomHeader)) {
+    const participantCount = getParticipantCount(roomHeader)
+    if (participantCount !== undefined) {
+      return `${participantCount} member${participantCount > 1 ? 's' : ''}`
+    }
+  }
+  return null
+}
+
+const ChatRoomHeader: FC<ChatRoomHeaderProps> = ({ roomHeaderRef }) => {
+  const roomHeader = useFragment(ChatRoomHeaderFragment, roomHeaderRef)
+
+  const isUpToMd = useResponsive('up', 'md')
+  const { resetChatRoom } = useChatRoom()
+
+  const { title, avatar } = useNameAndAvatar(roomHeaderRef)
```

Comment:
I've noticed that `useFragment` is being called twice for the same fragment in both the `ChatRoomHeader` component and the `useNameAndAvatar` hook. This results in the same fragment being read twice within the same render tree, introducing unnecessary complexity.

Passing the `roomHeader` object (resolved data from `useFragment`) directly to useNameAndAvatar would probably be the best choice here